polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -697,6 +750,26 @@ impl<H: Copy> From<CoreState<H>> for super::v8::CoreState<H> {
 	}
 }
 
+/// Returns a mapping between the para id and the core indices assigned at different
+/// depths in the claim queue.
+pub fn transpose_claim_queue(
+	claim_queue: BTreeMap<CoreIndex, VecDeque<Id>>,
+) -> BTreeMap<ParaId, BTreeMap<u8, BTreeSet<CoreIndex>>> {
```

Comment:
Some type alias should help here.

Reply:
Agree :)
Also, I find it a bit strange to have a `BTreeMap<u8, BTreeSet<_>>`
instead of a `Vec<BTreeSet<CoreIndex>>` where the index in the Vec represents the depth. Given that the depth is supposed to be a number 0..x and without gaps IIUC

Reply:
I was using a `VecDeque` initially, but switched to `BTreeMap`. There are 2 reasons to do it:
- we don't need to pop like with the claim. queue
- future proofing in case we want to support gaps for on-demand, where currently we need to double the assignment in case the core is not used at all, as the para might not have enough time to utilize it if we add it at claim queue offset 0